src/complex/Filter/IFilter.cpp

```diff
@@ -116,12 +116,25 @@ IFilter::PreflightResult IFilter::preflight(const DataStructure& data, const Arg
 IFilter::ExecuteResult IFilter::execute(DataStructure& data, const Arguments& args, const PipelineFilter* pipelineFilter, const MessageHandler& messageHandler) const
 {
   // determine required parameters
+  Parameters params = parameters();
+  Arguments resolvedArgs;
 
+  for(auto&& [name, arg] : args)
+  {
+    resolvedArgs.insert(name, arg);
+  }
   // substitute defaults
 
+  for(auto&& [name, parameter] : params)
```

Comment:
I think this change is ok for now. I've got an idea on how I want to factor the common argument handling in preflight/execute, but I haven't gotten to it yet.

Reply:
OK